### ISSUE TITLE
[BUG] trace 'retain-on-failure': is not saving the entire trace when one of steps fail

### ISSUE THREAD

**Issue by franciscamoretto** [microsoft/playwright#22964]
### System info
- Playwright Version: [v1.33.0]
- Operating System: [All, Windows 11, Ubuntu 20, macOS 13.2, etc.]
- Browser: [All, Chromium, Firefox, WebKit]
- Other info:

### Source code

- [x] I provided exact source code that allows reproducing the issue locally.

**Config file**

```js
// playwright.config.ts
import type { PlaywrightTestConfig } from '@playwright/test';
import { LaunchOptions, devices } from '@playwright/test';

/**
 * Read environment variables from file.
 * https://github.com/motdotla/dotenv
 */
require('dotenv').config();

const browserOptions: LaunchOptions = {
  slowMo: 0,
  args: [
    '--use-fake-ui-for-media-stream',
    '--use-fake-device-for-media-stream',
  ],
  firefoxUserPrefs: {
    'media.navigator.streams.fake': true,
    'media.navigator.permission.disabled': true,
  },
};

/**
 * See https://playwright.dev/docs/test-configuration.
 */
const config: PlaywrightTestConfig = {
  testDir: './tests',
  /* Maximum time one test can run for. */
  timeout: 500 * 1000,
  expect: {
    /**
     * Maximum time expect() should wait for the condition to be met.
     * For example in `await expect(locator).toHaveText();`
     */
    timeout: 50000,
  },
  /* Run tests in files in parallel */
  fullyParallel: process.env.PARALLEL == 'true',
  /* Fail the build on CI if you accidentally left test.only in the source code. */
  forbidOnly: !!process.env.CI,
  /* Retry on CI only */
  retries: parseInt(process.env.RETRIES || '1'),
  /* Opt out of parallel tests on CI. */
  workers: parseInt(process.env.WORKERS || '20'),
  /* Reporter to use. See https://playwright.dev/docs/test-reporters */
  reporter: [['html', { outputFolder: 'playwright-report' }]],
  /* Shared settings for all the projects below. See https://playwright.dev/docs/api/class-testoptions. */
  use: {
    baseURL: process.env.FRONT,
    headless: process.env.HEADLESS == 'true',
    launchOptions: browserOptions,
    trace: 'on',
    // browserName: process.env.BROWSER,
    channel: process.env.CHANNEL,
  },
  /* Configure projects for major browsers */
  projects: [
    {
      name: 'Chrome Desktop',
      use: {
        headless: true,
        ...devices['Desktop Chrome'],
        channel: 'chrome',
      },
    },
  ],

  /* Folder for test artifacts such as screenshots, videos, traces, etc. */
  // outputDir: 'test-results/',

  /* Run your local dev server before starting the tests */
  // webServer: {
  //   command: 'npm run start',
  //   port: 3000,
  // },
};

export default config;
```

---

**Steps**
1- Set the config `trace: 'retain-on-failure',` on // playwright.config.ts
2- Set the HTML report config `reporter: [['html', { outputFolder: 'playwright-report' }]],` on // playwright.config.ts
3- Create any test that contains at least 5 steps and one of the tests is failing
4- Observe the time-travel results 

---

Use this command to run the test and check the time travel trace

`npx playwright test`

or

`npx playwright test --reporter=html`

**Expected**

This is the expected behaviour when enabling the config `trace: 'retain-on-failure',`. Using this config it should display all steps in the time-travel report if any step failed.

https://github.com/microsoft/playwright/assets/83456381/5119f13d-7d8d-4ea5-b461-4d727a4b28a1

**Actual**

This is the behaviour when enabling the config `trace: 'retain-on-failure',`. Please see attached video

https://github.com/microsoft/playwright/assets/83456381/fa90f076-e2a2-4ea2-8ddb-421154858f8c

**Comment by franciscamoretto** [comment]
Hi @pavel share the repo is not necessary. The reason that you could not reproduce is that you just copied and paste the configs I posted. If you update according to the 1st step, you are able to reproduce. Thanks in advance

1- Set the config `trace: 'retain-on-failure'`, on // playwright.config.ts

**Comment by pavelfeldman** [comment]
I did that!

**Comment by franciscamoretto** [comment]
Ok I will prepare demo repo.

**Comment by franciscamoretto** [comment]
I just created a new repo and framework from scratch. Copied the package-lock.json, package.json, and the playwright.config to a new project I could not reproduce it either. If I manage to repo it in a demo scenario I'll open a new issue. Thanks

https://github.com/microsoft/playwright/assets/83456381/6c8e5cda-afa3-4944-be6b-0158572540b8

**Comment by pavelfeldman** [comment]
Closing as per above, please feel free to open a new issue if this does not cover your use case.
